src/modules/groups/groups.js

```diff
@@ -0,0 +1,93 @@
+/* eslint-disable indent */
+const express = require('express');
+const axios = require('axios');
+
+const router = express.Router();
+
+const getAccessToken = async () => {
+  const { CLIENT_SECRET, CLIENT_ID, DEFAULT_MS_GRAPH_SCOPE, MS_LOGIN_HOST_NAME, TENANT_ID } =
+    process.env;
+  try {
+    const URL = `${MS_LOGIN_HOST_NAME}/${TENANT_ID}/oauth2/v2.0/token`;
+    const { data } = await axios.post(
+      URL,
+      {
+        grant_type: 'client_credentials',
+        client_secret: CLIENT_SECRET,
+        client_id: CLIENT_ID,
+        scope: DEFAULT_MS_GRAPH_SCOPE
+      },
+      {
+        headers: {
+          'Content-Type': 'multipart/form-data'
+        }
+      }
+    );
+    return { accessToken: data?.access_token };
+  } catch (error) {
+    return { error };
+  }
+};
+
+router.get('/token', async (req, res) => {
+  const { accessToken, error } = (await getAccessToken()) || {};
+  if (error) {
+    return res.status(400).send({
+      error
+    });
+  }
+  return res.status(200).send({
+    accessToken
+  });
+});
+
+router.get('/all/:type', async (req, res) => {
+  const { type } = req.params;
+  const { MS_GRAPH_HOST_NAME, WRITERS_GROUP_ID, EDITOR_GROUP_ID, SIZING_GROUP_ID, ADMIN_GROUP_ID } =
+    process.env;
+  const { accessToken } = (await getAccessToken()) || {};
+  let groupId;
+
+  switch (type) {
+    case 'writers':
+      groupId = WRITERS_GROUP_ID;
+      break;
+    case 'editors':
+      groupId = EDITOR_GROUP_ID;
+      break;
+    case 'admin':
+      groupId = ADMIN_GROUP_ID;
+      break;
+    case 'sizing':
+      groupId = SIZING_GROUP_ID;
+      break;
+    default:
+      groupId = null;
+  }
+
+  if (!groupId) {
```

Comment:
This is not a validation logic. This is part of the api.
It should remain within this loop only.

Since there is no validation, i dont see any reason to move this to middleware.

Reply:
i have designed this to validate query, path or body and it will take care of these validations to make sure groups can only have writers,editors, etc

with this we no need to validate path .body or query prams in request handlers

Reply:
for ref: C:\Users\nikhil.s\apps\bc-cms-rewrite-node-ms\src\modules\workflows\dtos.js 
once u define here, apply here 
![image](https://github.com/raj-kumar-litmus/bc-cms-rewrite-node-ms/assets/132487558/750c92fe-e2a7-4d73-8486-6ed9c007f0ec)

Reply:
I dont we need to validate the incoming param.

 The switch case chooses the groupid which will be used in the subsequent Azure Api call.
I dont think we should move this logic to middleware.

Reply:
This, we agreed not to proceed since you werent sure about the JOI for enums.